bson/src/main/org/bson/codecs/pojo/ConventionAnnotationImpl.java

```diff
@@ -138,10 +138,11 @@ private void processPropertyAnnotations(final ClassModelBuilder<?> classModelBui
             for (int i = 0; i < properties.size(); i++) {
                 BsonProperty bsonProperty = properties.get(i);
                 Class<?> parameterType = parameterTypes.get(i);
+                TypeData<?> parameterTypeData = TypeData.builder(parameterType).build();
                 PropertyModelBuilder<?> propertyModelBuilder = classModelBuilder.getProperty(bsonProperty.value());
                 if (propertyModelBuilder == null) {
                     addCreatorPropertyToClassModelBuilder(classModelBuilder, bsonProperty.value(), parameterType);
-                } else if (propertyModelBuilder.getTypeData().getType() != parameterType) {
+                } else if (!propertyModelBuilder.getTypeData().equals(parameterTypeData)) {
```

Comment:
Can you explain how this change relates to the fix?  I'm not following.

Reply:
This ensures that we are comparing TypeData to TypeData, which handles the boxing of primitives.  Without it the ConventionAnnotation won't allow primitives to be used in the `@BsonCreator` methods.